code/game/objects/structures/holosign.dm

```diff
@@ -104,6 +104,62 @@
 	. = ..()
 	air_update_turf(TRUE)
 
+///this is a machinery to make it possible to check on the APC power to shut it down in case of an outage
+/obj/machinery/holosign/barrier/power_shield
+	name = "powered shield"
+	desc = "A shield to prevent changes of atmospheric and heat transfer"
+	icon = 'icons/effects/effects.dmi'
+	density = FALSE
+	anchored = TRUE
+	CanAtmosPass = ATMOS_PASS_NO
+	resistance_flags = FIRE_PROOF
+	///store the conductivity value of the turf is applyed so that it can be restored on removal
+	var/stored_conductivity = 0
+	///power drain from the apc, in W (so 5000 is 5 kW), per each holosign placed
+	var/power_consumption = 5000
+
+/obj/machinery/holosign/barrier/power_shield/Initialize()
+	. = ..()
+	air_update_turf(TRUE)
+	var/area/a = get_area(src)
+	a.addStaticPower(power_consumption, STATIC_EQUIP)
+	if(a.power_environ == TRUE)
+		shield_turf()
+	else
+		Destroy()
```

Comment:
You probably mean `qdel(src)` her or something.
You'll want to report this with a `stack_trace` or `CRASH`, though.

Reply:
How do I do a stack_trace report?

Reply:
`CRASH("Help, this happened!")` will `return` and create a runtime report with what your wrote there.
`stack_track("This happened!")` will produce a runtime without returning.

I assume you don't want this being deleted on initialize, if that happened then something is wrong.

Reply:
mm i made it that if there isnt power when initializing it would get destroyed

Reply:
Instead of creating and destroying it, make sure that doesn't happen in the first place and make it scream with the procs I gave you in case it does, so it gets fixed.

Reply:
aight

Reply:
ok i dont know how to use the initialize() properly to not placing the thing and then destroy it, i was thinking of placing `if(a.power_environ == FALSE)` before the . = ..() but i dont know if that works

Reply:
don't spawn it if it shouldnt be able to be spawned

Reply:
that's the point, how? i should stop the initialize if there is no power?

Reply:
Make sure the `Initialize()` doesn't happen in the first place.